src/main/java/org/jabref/gui/maintable/ColumnFactory.java

```diff
@@ -0,0 +1,185 @@
+package org.jabref.gui.maintable;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+import java.util.Objects;
+import java.util.Optional;
+import java.util.stream.Collectors;
+
+import javafx.scene.Node;
+import javafx.scene.control.TableColumn;
+
+import org.jabref.gui.IconTheme;
+import org.jabref.gui.JabRefIcon;
+import org.jabref.gui.externalfiletype.ExternalFileType;
+import org.jabref.gui.externalfiletype.ExternalFileTypes;
+import org.jabref.gui.specialfields.SpecialFieldValueViewModel;
+import org.jabref.gui.specialfields.SpecialFieldViewModel;
+import org.jabref.gui.util.ValueTableCellFactory;
+import org.jabref.model.database.BibDatabase;
+import org.jabref.model.entry.FieldName;
+import org.jabref.model.entry.LinkedFile;
+import org.jabref.model.entry.specialfields.SpecialField;
+
+import org.fxmisc.easybind.EasyBind;
+
+class ColumnFactory {
```

Comment:
I am not that happy with this generic name here... as this shadows the javafx classes 
Maybe MaintableColumnFactory? Same as with the other CellFactory